From 5377425b11c38c51a9209e55346bf8f0dee45cc0 Mon Sep 17 00:00:00 2001
From: Jiri Slaby <jslaby@suse.cz>
Date: Mon, 20 May 2019 07:48:13 +0200
Subject: [PATCH 2/4] cmake: set library VERSIONs

Especially soversion is needed to have a correctly linked and named
libraries.
---
 CMakeLists.txt | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 24d7e26..4ff736d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,15 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 2.8.10)
-PROJECT (DXFRW)
+
+if(${CMAKE_MAJOR_VERSION} LESS 3)
+	project(DXFRW)
+	set(VERSION 0.6.3)
+	set(PROJECT_VERSION 0.6.3)
+	set(PROJECT_VERSION_MAJOR 0)
+else()
+	cmake_policy(SET CMP0048 NEW)
+	project(DXFRW VERSION 0.6.3)
+endif()
+
 ADD_DEFINITIONS(-std=c++11)
 
 include(GNUInstallDirs)
@@ -69,8 +79,9 @@ foreach(dir ${dirs})
 endforeach()
 
 add_library(dxfrw SHARED ${libdxfrw_srcs} ${libdxfrw_hdrs})
+set_target_properties(dxfrw PROPERTIES VERSION "${PROJECT_VERSION}" SOVERSION "${PROJECT_VERSION_MAJOR}")
 
 # INSTALLATION
 install(TARGETS dxfrw DESTINATION "${CMAKE_INSTALL_LIBDIR}")
-set(HDRSUBDIR "/libdxfrw0/")
+set(HDRSUBDIR "/libdxfrw${PROJECT_VERSION_MAJOR}/")
 install(FILES ${installdxfrw_hdrs} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${HDRSUBDIR}")
-- 
2.21.0

